### ISSUE TITLE
Using packrat in multiple projects

### ISSUE THREAD

**Issue by JauntyJJS** [rstudio/packrat#469]
Hi,

I apologize in advanced if I am using packrat wrongly.
I hope that I explain the situation clearly.

I have used packrat in one project (project A) in one directory and save the packages used using snapshot()

For project B in another directory, I have unbundled another packrat project from a collaborator. 
It creates a new directory "CollabName" with the .Rprofile file, packrat folder. I then off the R gui and change the new directory name from "CollabName" to "MyName".

I then restarted R and set the working directory  to project B/MyName

I wanted to use snapshot to save this new project. Unfortunately, when I type status(), packrat remember the packages used in project A and but not the packages used in "MyName" for project B.

"The following packages are referenced in your code, but are not present
in your library nor in packrat:"

{The packages used in project A}

Is there a way to update status() to look at the package status for project B and not project A 
Thank you.

> sessionInfo()
R version 3.4.4 (2018-03-15)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 15063)

Matrix products: default

locale:
[1] LC_COLLATE=English_Singapore.1252  LC_CTYPE=English_Singapore.1252   
[3] LC_MONETARY=English_Singapore.1252 LC_NUMERIC=C                      
[5] LC_TIME=English_Singapore.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_3.4.4  tools_3.4.4     rtvs_1.0.0.0    packrat_0.4.9-1

**Comment by JauntyJJS** [comment]
I tried to reload all the packages used in Project A and load it in Project B using the restore() function. Now project B has the same packages as project A. 

Then I use the clean() function to forcefully remove the packages that I do not need in package B. I was able to do that. However, when I tried to snapshot Project B package updated status, it gives the error that a package used in Project A but not in Project B is missing

Error: Unable to retrieve package records for the following packages:
- "tibble"

Is there a way to update snapshot that the packages used for project B is different from project A.
I tried to even delete the information given in the packrat.lock file for Project B but it still does not work ?

I just want to inform packrat that the packages used in Project A and Project B are different and hopefully update snapshot to remember Project B packages instead of Project A.

Please advise on the correct procedure to do this ?

Thank you.

**Comment by kevinushey** [comment]
Can you share the `packrat.lock` file used in this project? Was `tibble` installed from CRAN / GitHub or another source?

**Comment by JauntyJJS** [comment]
Hi,

I am sorry but I have deleted both packrat folders in project A and project B and was unable to reproduce the problem...because a new error has just shows up. 

I have updated R and R tools to the latest version and wanted to recreate a packrat on an existing R script on ProjectA

[ProjectA.zip](https://github.com/rstudio/packrat/files/1953715/ProjectA.zip)

Unfortunately, this new error appears

![image](https://user-images.githubusercontent.com/9066508/39342499-f4d81f7a-4a0a-11e8-8b7e-9dfd1507fe3f.png)

Now I am unable to use packrat on project A ...

**Comment by JauntyJJS** [comment]
Hi, 

After some trial and error, I have solved the issue myself.
For the reason why packrat cannot initialize in project A was because when packrat sees an existing R script in the folder that it was asked to initialise, it will try to load the packages used in the script by using the default CRAN but fails because one of my packages "ComplexHeatMap" was from Bioconductor.

What I did was that I move the script from this folder, let packrat initialise an empty directory. That way, it will just install packrat. And then I move the script back to the folder and re install all the packages from scratch in the "packrat environment" and save it using snapshot and bundle it

For project B issue, it was because besides changing a directory name, I added a new R script but did not install the relevant packages that the newly added R script needs. Hence, when I type status, the error occurs that it could not find any records of packages that the new R script needs. I find that out by moving the new R script to the new location like what I did in Project A and the status did not give any error.

So in the end, I put the script back and install all the necessary packages and the snapshot works.
